test/librarySearch.lean

```diff
@@ -102,3 +102,12 @@ by library_search using P, Q -- exact P ∩ Q
 
 example (n : ℕ) (r : ℚ) : ℚ :=
 by library_search using n, r -- exact nsmulRec n r
+
+-- Check that we don't use sorryAx:
```

Comment:
I'm pretty sure that the test would still pass if we got `sorryAx`.

Reply:
Correct, the test will still pass. My hope was that the comment with the expected output would at least help someone notice the problem.
Do you know of a way to assert that a theorem does not use `sorry`?

Reply:
This is the best I can think of quickly:
```lean
import Lean
open Lean Meta

def foo : Nat := 42

#eval show MetaM Unit from do
  guard <| !(← getConstInfoDefn ``foo).value.hasSorry
```

Reply:
I added a new `assert_no_sorry` command. I tried `getConstInfoDef` as you suggested, but it failed to reliably find all sorries (I think some misdirection through a `proof_1` constant was confusing it). So I instead used `Lean.Elab.Command.CollectAxioms.collect`, which is the same machinery that `#print axioms` uses.